orderer/common/multichannel/registrar.go

```diff
@@ -129,94 +128,211 @@ func (r *Registrar) initializeJoinBlockFileRepo() {
 func (r *Registrar) Initialize(consenters map[string]consensus.Consenter) {
 	r.consenters = consenters
 
-	//TODO reconcile existing channel ledgers with join blocks
+	// Discover and load join-blocks. If there is a join-block, there must be a ledger; if there is none, create it.
+	// channelsWithJoinBlock maps channelID to a join-block.
+	channelsWithJoinBlock := r.loadJoinBlocks()
+
+	// Discover all ledgers. This should already include all channels with join blocks as well.
 	existingChannels := r.ledgerFactory.ChannelIDs()
 
-	// TODO first scan for and initialize the system channel, if it exists.
-	// TODO then initialize application channels, by creating either consensus.Chain or follower.Chain.
-	// TODO start all channels
+	// Scan for and initialize the system channel, if it exists.
+	// Make sure there are no empty ledgers without a corresponding join-block.
+	// Note that there may be channels with empty ledgers, but always with a join block.
+	r.initSystemChannel(existingChannels, channelsWithJoinBlock)
+
+	// Initialize application channels, by creating either a consensus.Chain or a follower.Chain.
+	r.initAppChannels(existingChannels, channelsWithJoinBlock)
+
+	// Start all channels
+	for _, chainSupport := range r.chains {
+		chainSupport.start()
+	}
+	for _, fChain := range r.followers {
+		fChain.Start()
+	}
+
+	if r.systemChannelID == "" {
+		logger.Infof("Registrar initializing without a system channel, number of application channels: %d", len(r.chains)+len(r.followers))
+		if _, etcdRaftFound := r.consenters["etcdraft"]; !etcdRaftFound {
```

Comment:
If it's much work, I'm certainly fine with skipping support of the Join API for solo and kafka.  My primary concern was that it would be impossible to run these consensus types without a system channel.

As a hypothetical user support case, we could have a network with a forked, or otherwise damaged/unrecoverable system channel.  We could delete the channel manually using `rm` on the filesystem, start the orderer, and then proceed to perform a Raft migration to get the network up and running under the new participation API.  Not saying it's something we want to test, or support, but it's nice to have these sorts of options when things invariably go wrong.